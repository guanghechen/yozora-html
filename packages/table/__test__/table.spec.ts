import type { Table } from '@yozora/ast'
import { renderChildren } from '../../../jest.setup'
import renderTable from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const node = {
      type: 'table',
      columns: [
        {
          align: 'center',
        },
        {
          align: 'right',
        },
        {
          align: null,
        },
      ],
      children: [
        {
          type: 'tableRow',
          children: [
            {
              type: 'tableCell',
              children: [
                {
                  type: 'text',
                  value: 'abc',
                },
              ],
            },
            {
              type: 'tableCell',
              children: [
                {
                  type: 'text',
                  value: 'defghi',
                },
              ],
            },
            {
              type: 'tableCell',
              children: [
                {
                  type: 'text',
                  value: 'xyz',
                },
              ],
            },
          ],
        },
        {
          type: 'tableRow',
          children: [
            {
              type: 'tableCell',
              children: [
                {
                  type: 'text',
                  value: 'bar',
                },
              ],
            },
            {
              type: 'tableCell',
              children: [
                {
                  type: 'text',
                  value: 'baz',
                },
              ],
            },
            {
              type: 'tableCell',
              children: [
                {
                  type: 'text',
                  value: 'defghi',
                },
              ],
            },
          ],
        },
      ],
    }
    expect(renderTable(node as Table, renderChildren)).toMatchSnapshot()
  })
})
