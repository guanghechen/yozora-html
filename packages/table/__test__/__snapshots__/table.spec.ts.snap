// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `"<table class="yozora-table-item"><thead class="yozora-table__thead"><tr class="yozora-table-row"><th class="yozora-table-cell" align="center"><span class="yozora-text">abc</span></th><th class="yozora-table-cell" align="right"><span class="yozora-text">defghi</span></th><th class="yozora-table-cell"><span class="yozora-text">xyz</span></th></tr></thead><tbody class="yozora-table__tbody"><tr class="yozora-table-row"><td class="yozora-table-cell" align="center"><span class="yozora-text">bar</span></td><td class="yozora-table-cell" align="right"><span class="yozora-text">baz</span></td><td class="yozora-table-cell"><span class="yozora-text">defghi</span></td></tr></tbody></table>"`;
